src/com/sahokia/kos/bustickets/validation/BusTicketsValidator.java

```diff
@@ -0,0 +1,129 @@
+package com.sahokia.kos.bustickets.validation;
+
+import com.sahokia.kos.bustickets.exception.PriceException;
+import com.sahokia.kos.bustickets.exception.StartDateException;
+import com.sahokia.kos.bustickets.exception.TicketClassException;
+import com.sahokia.kos.bustickets.exception.TicketTypeException;
+import com.sahokia.kos.bustickets.model.BusTicket;
+import com.sahokia.kos.bustickets.model.TicketClass;
+import com.sahokia.kos.bustickets.model.TicketType;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BusTicketsValidator {
+
+    public void validateBusTickets(List<BusTicket> busTickets) {
+        Map<String, Boolean> current;
+        int total = 0;
+        int valid = 0;
+        Map<String, Integer> violations = new HashMap<>();
+        violations.put("ticket class", 0);
+        violations.put("ticket type", 0);
+        violations.put("price", 0);
+        violations.put("start date", 0);
+        for (BusTicket busTicket : busTickets) {
+            total++;
+            current = validateBusTicket(busTicket);
+            if (!current.containsValue(true)) {
+                valid++;
+            } else {
+                current.forEach((key, value) -> {
+                    if (value) {
+                        violations.put(key, violations.get(key) + 1);
+                    }
+                });
+            }
+        }
+        System.out.println("Total = " + total);
+        System.out.println("Valid = " + valid);
+        System.out.println("Most popular violation = " + violations.entrySet()
+                .stream()
+                .max(Map.Entry.comparingByValue())
+                .map(Map.Entry::getKey)
+                .orElse(null));
+    }
+
+    public Map<String, Boolean> validateBusTicket(BusTicket busTicket) {
+        Map<String, Boolean> violations = new HashMap<>();
+        violations.put("ticket class", false);
```

Comment:
"ticket class" and similar we could put as variables to avoid magic Strings

Reply:
Removed these magic strings to constants.